src/components/icon/index.tsx

```diff
@@ -54,12 +65,48 @@ const Icon = forwardRef((props: Props, ref: any) => {
     assetName,
     modifiers,
     recorderTag,
+    badgeProps,
+    hideBadge,
     ...others
   } = props;
   const {margins} = modifiers;
+  const shouldShowBadge = !hideBadge && badgeProps;
   const iconSize = size ? {width: size, height: size} : undefined;
   const shouldFlipRTL = supportRTL && Constants.isRTL;
 
+  const getBadgeStyling = (): StyleProp<ViewStyle> => {
+    const containerStyle = badgeProps?.containerStyle;
+    const badgePosition: StyleProp<ViewStyle> = {
+      position: 'absolute',
+      top: -3
```

Comment:
why did you set it to -3 if you're changing it later? I'd set the default here, for the badge, and if it is a pimple override.